server/src/main/java/org/opensearch/cluster/metadata/ComposableIndexTemplate.java

```diff
@@ -264,51 +266,74 @@ public String toString() {
 
     public static class DataStreamTemplate implements Writeable, ToXContentObject {
 
-        private static final ObjectParser<DataStreamTemplate, Void> PARSER = new ObjectParser<>(
+        private static final ParseField TIMESTAMP_FIELD_FIELD = new ParseField("timestamp_field");
+
+        private static final ConstructingObjectParser<DataStreamTemplate, Void> PARSER = new ConstructingObjectParser<>(
             "data_stream_template",
-            DataStreamTemplate::new
+            true,
+            args -> new DataStreamTemplate((TimestampField) args[0])
         );
 
+        static {
+            PARSER.declareObject(ConstructingObjectParser.optionalConstructorArg(), TimestampField.PARSER, TIMESTAMP_FIELD_FIELD);
+        }
+
+        private final TimestampField timestampField;
+
         public DataStreamTemplate() {
+            this(DataStreamFieldMapper.Defaults.TIMESTAMP_FIELD);
+        }
+
+        public DataStreamTemplate(TimestampField timestampField) {
+            this.timestampField = timestampField;
         }
 
-        public String getTimestampField() {
-            return FIXED_TIMESTAMP_FIELD;
+        public DataStreamTemplate(StreamInput in) throws IOException {
+            this.timestampField = in.readOptionalWriteable(TimestampField::new);
```

Comment:
This is a new (optional) field in OpenSearch 1.0.0 for a class that existed in previous versions. To support backwards compatibility we need to wrap in a version check.
```suggestion
            if (in.getVersion().onOrAfter(Version.V_1_0_0)) {
                this.timestampField = in.readOptionalWriteable(TimestampField::new);
            }
```

Reply:
Done.